pkg/columns/columns.go

```diff
@@ -0,0 +1,262 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package columns
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"sort"
+	"strings"
+)
+
+type ColumnMap[T any] map[string]*Column[T]
+
+type Columns[T any] struct {
+	// columns map[string]*Column[T]
+	ColumnMap[T]
+	options *Options
+}
+
+const virtualIndex = -1
+
+var stringType = reflect.TypeOf("") // used for virtual columns and columns with a custom extractor
+
+// MustCreateColumns creates a new column helper and panics if it cannot successfully be created; useful if you
+// want to initialize Columns as a global variable inside a package (similar to regexp.MustCompile)
+func MustCreateColumns[T any](options ...Option) *Columns[T] {
+	cols, err := NewColumns[T](options...)
+	if err != nil {
+		panic(err)
```

Comment:
Is it possible to specify in the error message the type for which this function failed?